test/Microsoft.AspNetCore.SignalR.Client.FunctionalTests/HubConnectionTests.cs

```diff
@@ -146,6 +148,81 @@ public HubConnectionTests()
 
         [Theory]
         [MemberData(nameof(HubProtocols))]
+        public async Task DynamicHubFixedMessage(IHubProtocol protocol)
+        {
+            var loggerFactory = CreateLogger();
+            const string originalMessage = "SignalR";
+
+            var httpConnection = new HttpConnection(new Uri("http://test/dynamic"), TransportType.LongPolling, loggerFactory, _testServer.CreateHandler());
+            var connection = new HubConnection(httpConnection, protocol, loggerFactory);
+            try
+            {
+                await connection.StartAsync();
+
+                var result = await connection.InvokeAsync<string>(nameof(DynamicTestHub.Echo), originalMessage);
+
+                Assert.Equal(originalMessage, result);
+            }
+            finally
+            {
+                await connection.DisposeAsync();
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(HubProtocols))]
+        public async Task DynamicHubCanReceiveMessage(IHubProtocol protocol)
+        {
+            var loggerFactory = CreateLogger();
+            const string originalMessage = "Hello";
+
+            var httpConnection = new HttpConnection(new Uri("http://test/dynamic"), TransportType.LongPolling, loggerFactory, _testServer.CreateHandler());
+            var connection = new HubConnection(httpConnection, protocol, loggerFactory);
+            try
+            {
+                await connection.StartAsync();
+
+                var tcs = new TaskCompletionSource<string>();
+                connection.On<string>("SendHello", tcs.SetResult);
+
+                await connection.InvokeAsync(nameof(DynamicTestHub.SendHello)).OrTimeout();
+
+                Assert.Equal(originalMessage, await tcs.Task.OrTimeout());
+            }
+            finally
+            {
+                await connection.DisposeAsync().OrTimeout();
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(HubProtocols))]
+        public async Task DynamicHubCanSendAndReceiveMessage(IHubProtocol protocol)
+        {
+            var loggerFactory = CreateLogger();
+            const string originalMessage = "Hello";
+
+            var httpConnection = new HttpConnection(new Uri("http://test/dynamic"), TransportType.LongPolling, loggerFactory, _testServer.CreateHandler());
+            var connection = new HubConnection(httpConnection, protocol, loggerFactory);
+            try
+            {
+                await connection.StartAsync();
+
+                var tcs = new TaskCompletionSource<string>();
+                connection.On<string>("SendMessage", tcs.SetResult);
```

Comment:
Change the name of the client (or server) method. I was a little confused at first by these being the same.